include/drivers/audio/pdm.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2020 Oticon A/S.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public PDM Driver APIs
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PDM_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PDM_H_
+
+/**
+ * @brief PDM Interface
+ * @defgroup pdm_interface PDM Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <errno.h>
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @typedef pdm_data_handler_t
+ * @brief User data handler for PDM samples
+ *
+ * @param data      Pointer to the PDM sample
+ * @param data_size Size of PDM sample
+ */
+typedef void (*pdm_data_handler_t)(int16_t *data,
+					uint16_t data_size);
+
+struct pdm_config {
+	/* Application data handler */
+	pdm_data_handler_t	data_handler;
+	/* Memory slab used for the released buffers */
+	struct k_mem_slab	*mem_slab;
+};
```

Comment:
I think it should be possible to configure more parameters here, like sampling frequency, mono/stereo selection, sample width, etc. See [i2s_config](https://github.com/zephyrproject-rtos/zephyr/blob/master/include/drivers/i2s.h#L299).